backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,268 @@
+                if workflow['status'] == "COMPLETED" and devId == aset.did:
+                    logger.debug("decommission device workflow completed")
+                    break
+            logger.debug("waiting for decommission device workflow...")
+            time.sleep(1)
+        else:
+            assert False, "decommission device workflow didn't complete for [%s]" % (aset.did,)
+
+        # check device gone from inventory
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = inventorym.with_auth(utoken).call('GET',
+                                            inventory.URL_DEVICE,
+                                            path_params={'id': aset.did})
+            if r.status_code == 404:
+                break
+            else:
+                logger.debug("waiting for the device to be removed from inventory...")
+                time.sleep(1)
+        else:
+            assert False, "device not removed from the inventory"
+
+        # check device gone from deviceauth
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = devauthm.with_auth(utoken).call('GET',
+                              deviceauth_v2.URL_DEVICE.format(id=aset.did))
+            if r.status_code == 404:
+                break
+            else:
+                logger.debug("waiting for the device to be removed from deviceauth...")
+                time.sleep(1)
+        else:
+            assert False, "device not removed from the deviceauth"
+
+class TestDeviceDecomissioning(TestDeviceDecomissioningBase):
+    def test_ok(self, user, devices):
+        self.do_test_ok(user, devices[0])
+
+class TestDeviceDecomissioningEnterprise(TestDeviceDecomissioningBase):
+    def test_ok(self, tenants_users_devices):
+        t = tenants_users_devices[0]
+        self.do_test_ok(user=t.users[0], device=t.devices[0], tenant_token=t.tenant_token)
+
+        t1 = tenants_users_devices[1]
+        self.verify_devices_unmodified(t1.users[0], t1.devices)
+
+    def verify_devices_unmodified(self, user, in_devices):
+        devauthm = ApiClient(deviceauth_v2.URL_MGMT)
+        useradmm = ApiClient(useradm.URL_MGMT)
+
+        r = useradmm.call('POST',
+                          useradm.URL_LOGIN,
+                          auth=(user.name, user.pwd))
+        assert r.status_code == 200
+
+        utoken = r.text
+
+        r = devauthm.with_auth(utoken).call('GET',
+                                            deviceauth_v2.URL_DEVICES)
+        assert r.status_code == 200
+        api_devs = r.json()
+
+        assert len(api_devs) == len(in_devices)
+        for ad in api_devs:
+            assert ad['status'] == 'pending'
+
+def parse_workflow_input(workflow_input):
+    d = {}
+    workflow_input = workflow_input.replace("{", "").replace("}", "").replace(" ", "")
+    for x in workflow_input.split(','):
+        k, v = x.split('=')
+        d[k] = v
+    return d
```

Comment:
This helper looks complicated :smiley: 

Maybe we can hide it in `conductor.py`? I think it belongs there and it could potentially be used by other tests